geometry/create_capacitance_file_theta.py

```diff
@@ -0,0 +1,381 @@
+    if line_color_number > 8:
+        line_color_number += 10
+    else:
+        line_color_number += 1
+    line_style_number += 1
+
+cTrace = TCanvas("cTrace","",600,400)
+cShield = TCanvas("cShield","",600,400)
+cDetector = TCanvas("cDetector","",600,400)
+
+legend = TLegend(0.1,0.693,0.8,0.9)
+#legend = TLegend(0.135,0.693,0.8,0.892)
+legend.SetHeader("Longitudinal layers")
+legend.SetNColumns(4)
+capa_shield_max = 0
+capa_det_max = 0
+for i in range (0, len(readoutLayerParallelLengths)):
+    print("--------------")
+    for index in range(0, numTheta):
+        theta = minTheta + index * deltaTheta
+        thetaNext = minTheta + (index+1) * deltaTheta
+        eta = -log(tan(theta/2.0))
+        deltaEta = abs( -log(tan((minTheta + (index+1) * deltaTheta)/2.0)) - eta)
+        if (debug):
+            print("theta = ", theta)
+            print("eta = ", eta)
+            print("delta eta = ", deltaEta)
+                         
+        # take into account the inclination in eta
+        #traceLength = trace_length[i] / (sin(2. * atan(exp(-eta))))
+        traceLength = trace_length[i] / sin(theta)
+        #print("Layer %d trace length %f"%(i+1, traceLength))
+        #Trace capacitance (stripline)
+        logStripline = log(3.1 * hs / (0.8 * w + t))
+        capacitanceTrace = nmult_trace * 1 / inch2mm * 1.41 * epsilonR / logStripline * traceLength
+        hCapTrace[i].SetBinContent(index+1, capacitanceTrace)
+    
+        #Shield capacitance (microstrip)
+        #cellLength = readoutLayerParallelLengths[i] / (sin(2. * atan(exp(-eta))))
+        cellLength = readoutLayerParallelLengths[i] / sin(theta)
+        logMicrostrip = log(5.98 * hm / (0.8 * ws + t))
+        # analytical formula 
+        #capacitanceShield = nmult * cellLength * tracesPerLayer[i] * 1 / inch2mm * 0.67 * (epsilonR + 1.41) / logMicrostrip
+        # from maxwell 
+        capacitanceShield = nmult * cellLength * tracesPerLayer[i] * capa_per_mm
+        if i == 1: #strip layer has smaller capacitance due to traces running beneath the anti-etch
+            capacitanceShield = nmult * cellLength * tracesPerLayer[i] * capa_per_mm_stripLayer
+        if capacitanceShield > capa_shield_max:
+            capa_shield_max = capacitanceShield
+        hCapShield[i].SetBinContent(index+1, capacitanceShield)
+
+        ##Detector area (C = epsilon*A/d)
+        #area = ( radius[i] * ( 1 / (tan(2. * atan(exp(- (index + 1) * deltaEta)))) -  1 / (tan(2. * atan(exp(- index * deltaEta))) ) )
+        #         + radius[i + 1] * ( 1 / (tan(2. * atan(exp(- (index + 1) * deltaEta)))) -  1 / (tan(2. * atan(exp(- index * deltaEta))) ) )
+        #         ) / 2. * (radius[i+1] - radius[i])
+        #distance = (radius[i+1] + radius[i]) / 2. * pi / Nplanes * cos (angle) - pcbThickness / 2. - passiveThickness / 2.
+
+        #Detector area (C = epsilon*A/d)
+        area = abs( real_radial_separation[i] * ( 1 / tan(thetaNext) -  1 / tan(theta) )
```

Comment:
If we go for the treatment proposed above, the area should be the smallest cell from the simulation, before merging, which will probably be the "strip cell size" (here it was the 'nominal cell' area and a division was applied for the strip layer).

Reply:
changed